api/v1alpha3/awsmachine_types.go

```diff
@@ -108,11 +108,15 @@ type CloudInit struct {
 	// the userdata from Secrets Manager and additionally delete the secret.
 	InsecureSkipSecretsManager bool `json:"insecureSkipSecretsManager,omitempty"`
 
-	// SecretARN is the Amazon Resource Name of the secret. This is stored
+	// SecretCount is the number of secrets used to form the complete secret
+	// +optional
+	SecretCount int32 `json:"secretCount,omitempty"`
```

Comment:
@detiber Were you suggesting this should be a pointer?

Reply:
original comment:
> I'm wondering if we need to add some type of validation here that the SecretCount != 0... Or do we think it would be valid to have a SecretCount of 0?

Reply:
I don't think we can consider a SecretCount of 0 as valid, and treating it as a "default" or "unset" value should be fine :)

Reply:
Added validations as such.